### PR TITLE
Remove rule CA1801 - replaced by IDE0060

### LINKED ISSUE #4498: Decide which analyzer to keep: CA1801 vs IDE0060
### Analyzer

**Diagnostic ID**: [CA1801](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1801)

### Describe the improvement

Currently 2 rules compete and they have both different implementation and "scope" (findings and issues), rather than "wasting" time maintaining both I would suggest that a clear cut is made and only one is kept. When the choice is made, we should ensure that the winning rule is correctly matching both scopes.

Tagging @mavasani

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4498

### CODE REVIEW

**Comment by Youssef1313** [comment]
There are open issues regarding some false positives.

https://github.com/dotnet/roslyn-analyzers/issues/3752
https://github.com/dotnet/roslyn-analyzers/issues/4342

We need to check whether these false positives exist in IDE0060 or not. Then close them or move to roslyn depending on whether IDE0060 has the same issues or not.

**Comment by Evangelink** [comment]
@mavasani It's written not to do it manually but there was no way to apply the code-fix for the removed rule so I did it manually. If that's not expected we should create a ticket.

**Comment by Youssef1313** [comment]
> @mavasani It's written not to do it manually but there was no way to apply the code-fix for the removed rule so I did it manually. If that's not expected we should create a ticket.

I think the codefix should be able to do that. If it doesn't, then it's a codefix issue.

**Comment by Evangelink** [comment]
Well the issue is only available in the error-list and from there we cannot run a code-fix so I think that's a more global topic (to run a code-fix from the error window).

**Comment by Youssef1313** [comment]
@Evangelink Oh correct. The diagnostic in this case doesn't have a location ;-(

File: src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/Maintainability/CSharpReviewUnusedParameters.Fixer.cs [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeQuality.Analyzers.Maintainability;
-
-namespace Microsoft.CodeQuality.CSharp.Analyzers.Maintainability
-{
-    /// <summary>
-    /// CA1801: Review unused parameters
-    /// </summary>
-    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
-    public sealed class CSharpReviewUnusedParametersFixer : ReviewUnusedParametersFixer
```
Comment (mavasani) [inline src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/Maintainability/CSharpReviewUnusedParameters.Fixer.cs:15]: IDE0060 does not have a code fix. CodeFix for CA1801 had some shortcomings and heuristics. I think we should consider opening a Roslyn issue to author a code fix for IDE0060, which uses this code fix implementation as a reference.

**Review by mavasani — APPROVED** [review]
Thank you @Evangelink. Please make sure to open a doc PR to mention deprecation once this is merged.

**Comment by mavasani** [comment]
> Well the issue is only available in the error-list and from there we cannot run a code-fix so I think that's a more global topic (to run a code-fix from the error window).

CA1801 should have a code fix from editor, the diagnostic does have a source file location. Is that not what you are observing?

**Comment by mavasani** [comment]
> It's written not to do it manually but there was no way to apply the code-fix for the removed rule so I did it manually

Ah, you mean the diagnostic to update the analyzer releases file? Sorry, I misunderstood. In that case, yes, that is by design as that diagnostic cannot be mapped to any source location. Manual update is fine for rule removal.

**Review by mavasani — APPROVED** [review]

**Comment by Evangelink** [comment]
@mavasani deprecation doc is already done and available live (see https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1801)

**Pull request merged**